docker_manager.py

```
@@ -0,0 +1,304 @@
+                        }
+                    }
+                )
+                
+                # Получаем внешний порт
+                container.reload()
+                port_info = container.attrs['NetworkSettings']['Ports'].get('8080/tcp')
+                if port_info:
+                    external_port = port_info[0]['HostPort']
+                    url = f"http://localhost:{external_port}"
+                else:
+                    url = f"http://localhost:8080"
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)